tests/e2e/mgmt_wrkld_domain_isolation_negative.go

```diff
@@ -0,0 +1,485 @@
+
+		ginkgo.By("Creating statefulset")
+		sts := createCustomisedStatefulSets(ctx, client, namespace, true, replicas, false, nil,
+			false, true, "", "", storageclassZ2, storagePolicyNameZ2)
+		allowedTopologies = setSpecificAllowedTopology(allowedTopologies, topkeyStartIndex, 1,
+			2)
+		err = verifyPvcAnnotationPvAffinityPodAnnotationInSvc(ctx, client, sts, nil, nil, namespace,
+			allowedTopologies)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		ginkgo.By("Mark zone-2 for removal from wcp namespace")
+		err = markZoneForRemovalFromWcpNs(vcRestSessionId, namespace,
+			zone2)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		// get ds name
+		pv := getPvFromClaim(client, namespace, pvclaim1.Name)
+		volHandle := pv.Spec.CSI.VolumeHandle
+		s1 := rand.NewSource(time.Now().UnixNano())
+		r1 := rand.New(s1)
+		randomStr := strconv.Itoa(r1.Intn(1000))
+		datastoreName, dsRef, _ := e2eVSphere.fetchDatastoreNameFromDatastoreUrl(ctx, volHandle)
+
+		ginkgo.By("Stop Vsan-health service")
+		ginkgo.By(fmt.Sprintln("Stopping vsan-health on the vCenter host"))
+		isVsanHealthServiceStopped = true
+		err = invokeVCenterServiceControl(ctx, stopOperation, vsanhealthServiceName, vcAddress)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		defer func() {
+			if isVsanHealthServiceStopped {
+				ginkgo.By(fmt.Sprintln("Starting vsan-health on the vCenter host"))
+				err := invokeVCenterServiceControl(ctx, startOperation, vsanhealthServiceName, vcAddress)
+				gomega.Expect(err).NotTo(gomega.HaveOccurred())
+				ginkgo.By(fmt.Sprintf("Sleeping for %v seconds to allow vsan-health to come up again", vsanHealthServiceWaitTime))
+				time.Sleep(time.Duration(vsanHealthServiceWaitTime) * time.Second)
+			}
+		}()
+
+		ginkgo.By("Rename datastore to a new name")
+		framework.Logf("Original datastore name: %s", datastoreName)
+		e2eVSphere.renameDs(ctx, datastoreName+randomStr, &dsRef)
+		defer func() {
+			framework.Logf("Renaming datastore back to original name")
+			e2eVSphere.renameDs(ctx, datastoreName, &dsRef)
+		}()
+
+		ginkgo.By("Start Vsan-health service on the vCenter host")
+		err = invokeVCenterServiceControl(ctx, startOperation, vsanhealthServiceName, vcAddress)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		ginkgo.By(fmt.Sprintf("Sleeping for %v seconds to allow vsan-health to come up again", vsanHealthServiceWaitTime))
+		time.Sleep(time.Duration(vsanHealthServiceWaitTime) * time.Second)
+		isVsanHealthServiceStopped = false
+
+		ginkgo.By("Perform scaling operation on statefulset. Increase the replica count to 4 when zone is marked for removal")
+		ginkgo.By("Scaleup any one StatefulSets replica")
+		_, scaleupErr := scaleStatefulSetPods(client, sts, 4)
+		gomega.Expect(scaleupErr).NotTo(gomega.HaveOccurred())
+
+		// Validate pods are in Pending state
+		pods, err := fss.GetPodList(ctx, client, sts)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		for _, pod := range pods.Items {
+			if pod.Name == "web-3" {
```

Comment:
Giving hard coded name i don't suggest here and why are you just comparing web-3, read all pods and check its status, that should suffice right